common/authenticator-requests/src/v5/registration.rs

```diff
@@ -170,15 +168,13 @@ impl GatewayClient {
     // Client should perform this step when generating its payload, using its own WG PK
     #[cfg(feature = "verify")]
     pub fn verify(&self, gateway_key: &PrivateKey, nonce: u64) -> Result<(), Error> {
-        // convert from 1.0 x25519-dalek private key into 2.0 x25519-dalek
-        #[allow(clippy::expect_used)]
-        let static_secret = x25519_dalek::StaticSecret::from(gateway_key.to_bytes());
+        let client_pubkey = PublicKey::from(*self.pub_key.deref());
 
-        let dh = static_secret.diffie_hellman(&self.pub_key);
+        let dh = gateway_key.diffie_hellman(&client_pubkey);
```

Comment:
```suggestion
        let dh = gateway_key.diffie_hellman(&self.pub_key)
```

Reply:
`gateway_key` is of type `nym_crypto::asymmetric::encryption::PrivateKey` and the self.pub_key istype `nym_wireguard_types::PeerPublicKey`. Those two don't have a direct way to do a dh against each other despite both wrapping dalek keys. So the option is converting or using a reference. Or adding more interop code to those types.

```
// convert public key to appropriate type
let client_pubkey = PublicKey::from(*self.pub_key.deref());
let dh = gateway_key.diffie_hellman(&client_pubkey);


// use gateways key as a ref to an x25519_dalek key
let dh = (gateway_key.as_ref()).diffie_hellman(&self.pub_key)

```

So I can change to using the second option there, but i think they are effectively pretty similar.